cache/grpcproxy/readcloser.go

```diff
@@ -0,0 +1,57 @@
+package grpcproxy
+
+import "io"
+
+type StreamReadCloser[M DataMessage] struct {
```

Comment:
Introducing generics to bazel-remote means that I need to find some time to do some background reading before I can understand this code...

Reply:
The generic is not really necessary here since we only use it with one message (`bs.ReadResponse`), I just found it more elegant and easier to reuse. I can remove if you feel this introduces an unwanted precendent.